docs/troubleshooting-guide/performance.md

```diff
@@ -0,0 +1,110 @@
+
+**How to detect this:**
+
+1. Run the same SQL that Metabase is running but using a different tool. For example, if you have created a question using the Notebook Editor and you are using Postgres as your database, you can view the SQL, copy it, and run it from the command line using `psql`. (If you have written the question in SQL you can just copy and paste it.) If the query runs noticeably faster this way than it does when you run it through Metabase, the problem is almost certainly one of the ones described above.
+
+2. If the query runs slowly when sent by another tool, see if you can write a SQL query that calculates the same result as the question you have built in Metabase, but does so more quickly.
+
+**How to fix this:* If the problem is the SQL we generate:
+
+1. Check if you have the most recent version of Metabase: we fix problems as they're reported, and updating Metabase may make your problem go away.
+2. You can use your SQL in place of the code we generate, and make its result available to people who prefer the Notebook Editor as a starting point for their questions. [LINK]
+3. And please file a bug report to help us figure out how to generate better SQL.
+
+### Values are repeatedly being converted on the fly.
+
+Low performance when using Metabase can also be caused by incorrect typing of columns, e.g., by storing a numeric value as a string.  When this happens, the query converts values on the fly each time the query is run.
+
+**How to detect this:** Even a handwritten query will be slow if it has to do this conversion every time. You can also spot this by looking at the raw data types of the columns being used in the query.
+
+**How to fix this:** Amend the database schema to store numbers as numbers, timestamps as timestamps, and so on, rather than as strings or other data types.
+
+### The Java Virtual Machine is using too much memory.
+
+We discuss this problem and its solutions in [this troubleshooting guide](./running.md).
+
+### Metabase is running on an under-powered machine.
+
+**How to detect this:** Checking the performance logs for the server where Metabase is running will tell you whether it is hitting CPU or memory limits. However, it's much more likely that the database itself is hitting its limits, so please check it first.
+
+**How to fix this:** Upgrade to a more powerful server, or one with more memory.
+
+Note: If you are running in the cloud, the servers we select are enough for 98% of customers: we use a 2-node cluster for all our in-house analytics and a 3-node cluster for customers that autoscales whenever performance requires it. If you are using our cloud option and the Metabase server really is the problem, you should consider deploying your own instance.
```

Comment:
I'm unsure if this note is even needed. If someone is using Metabase Cloud and experiencing problems, then it's likely some different problems than general "slowness". And would require more debugging from our side, since it's a managed service, so people would never know if they're hitting CPU or memory limits.

Reply:
incorporated